@codex

```
[Test (ubuntu-latest, stable)](https://github.com/benjisho/windows-mtr/actions/runs/23033259352/job/66895834902?pr=94#logs)
failed 3 minutes ago in 1m 8s
Search logs
1s
2s
5s
7s
17s
2s
31s
Run cargo test --all
   Compiling libc v0.2.183
   Compiling once_cell v1.21.4
   Compiling zerocopy v0.8.42
   Compiling tracing-core v0.1.36
   Compiling tracing v0.1.44
   Compiling anstyle-parse v1.0.0
   Compiling jobserver v0.1.34
   Compiling anstream v1.0.0
   Compiling getrandom v0.2.17
   Compiling mio v1.1.1
   Compiling cc v1.2.56
   Compiling rand_core v0.6.4
   Compiling parking_lot_core v0.9.12
   Compiling parking_lot v0.12.5
   Compiling socket2 v0.6.3
   Compiling errno v0.3.14
   Compiling signal-hook-registry v1.4.8
   Compiling clap_lex v1.1.0
   Compiling tokio v1.50.0
   Compiling clap_builder v4.6.0
   Compiling signal-hook v0.3.18
   Compiling nix v0.29.0
   Compiling ppv-lite86 v0.2.21
   Compiling rand_chacha v0.3.1
   Compiling rand v0.8.5
   Compiling clap v4.6.0
   Compiling signal-hook-mio v0.2.5
   Compiling caps v0.5.6
   Compiling tracing-serde v0.2.0
   Compiling tracing-log v0.2.0
   Compiling tracing-subscriber v0.3.22
   Compiling trippy-privilege v0.13.0
   Compiling crossterm v0.28.1
   Compiling hickory-proto v0.24.4
   Compiling dns-lookup v2.1.1
   Compiling socket2 v0.5.10
   Compiling roff v1.1.0
   Compiling ratatui v0.29.0
   Compiling trippy-core v0.13.0
   Compiling clap_mangen v0.2.33
   Compiling tracing-chrome v0.7.2
   Compiling hickory-resolver v0.24.4
   Compiling trippy-dns v0.13.0
   Compiling clap_complete v4.6.0
   Compiling trippy-tui v0.13.0
   Compiling zstd-sys v2.0.16+zstd.1.5.7
   Compiling bzip2-sys v0.1.13+1.0.8
   Compiling lzma-sys v0.1.20
   Compiling password-hash v0.4.2
   Compiling trippy v0.13.0
   Compiling pbkdf2 v0.11.0
   Compiling windows-mtr v1.1.3 (/home/runner/work/windows-mtr/windows-mtr)
   Compiling zstd-safe v5.0.2+zstd.1.5.2
   Compiling bzip2 v0.4.4
   Compiling xz2 v0.1.7
   Compiling zstd v0.11.2+zstd.1.5.2
   Compiling zip v0.6.6
   Compiling xtask v0.1.0 (/home/runner/work/windows-mtr/windows-mtr/xtask)
    Finished `test` profile [unoptimized + debuginfo] target(s) in 24.89s
     Running unittests src/main.rs (target/debug/deps/mtr-dcf6770cba808297)

running 16 tests
test tests::build_embedded_trippy_args_applies_enhanced_defaults ... ok
test tests::build_embedded_trippy_args_maps_core_flags ... ok
test tests::build_embedded_trippy_args_supports_json_mode ... ok
test tests::parse_passthrough_flags_allows_literal_apostrophe_values ... ok
test tests::parse_passthrough_flags_preserves_inner_quoted_values ... ok
test tests::parse_passthrough_flags_rejects_invalid_shell_quoting ... ok
test tests::parse_passthrough_flags_splits_single_quoted_token ... ok
test tests::should_not_print_banner_for_json_modes ... ok
test tests::verify_options_accepts_valid_ports ... ok
test tests::should_print_banner_for_non_json_modes ... ok
test tests::validate_target_accepts_ip_and_hostname ... ok
test tests::validate_target_rejects_invalid_target ... ok
test tests::verify_options_rejects_conflicting_enhanced_passthrough ... ok
test tests::verify_options_rejects_enhanced_threshold_ordering_errors ... ok
test tests::verify_options_rejects_report_wide_without_report_mode ... ok
test tests::verify_options_requires_port_for_tcp_udp ... ok

test result: ok. 16 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests/cli_tests.rs (target/debug/deps/cli_tests-504c37a99d397424)

running 5 tests
test test_basic_execution ... ignored
test test_version_output ... ok
test test_enhanced_wrapper_conflicts_with_passthrough_override ... FAILED
test test_enhanced_ui_conflicts_with_report_mode ... ok
test test_help_output ... ok

failures:

---- test_enhanced_wrapper_conflicts_with_passthrough_override stdout ----

thread 'test_enhanced_wrapper_conflicts_with_passthrough_override' (4213) panicked at tests/cli_tests.rs:65:5:
assertion failed: stderr.contains("--trippy-flags cannot override windows-mtr enhanced UI wrapper settings")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_enhanced_wrapper_conflicts_with_passthrough_override

test result: FAILED. 3 passed; 1 failed; 1 ignored; 0 measured; 0 filtered out; finished in 5.98s

error: test failed, to rerun pass `-p windows-mtr --test cli_tests`
Error: Process completed with exit code 101.
```

```
[Test (windows-latest, stable)](https://github.com/benjisho/windows-mtr/actions/runs/23033259352/job/66895834915?pr=94#logs)
failed 2 minutes ago in 2m 25s
Search logs
2s
8s
27s
17s
34s
2s
50s
test tests::build_embedded_trippy_args_applies_enhanced_defaults ... ok
test tests::build_embedded_trippy_args_maps_core_flags ... ok
test tests::parse_passthrough_flags_allows_literal_apostrophe_values ... ok
test tests::parse_passthrough_flags_preserves_inner_quoted_values ... ok
test tests::parse_passthrough_flags_rejects_invalid_shell_quoting ... ok
test tests::parse_passthrough_flags_splits_single_quoted_token ... ok
test tests::should_not_print_banner_for_json_modes ... ok
test tests::should_print_banner_for_non_json_modes ... ok
test tests::verify_options_accepts_valid_ports ... ok
test tests::verify_options_rejects_conflicting_enhanced_passthrough ... ok
test tests::verify_options_rejects_enhanced_threshold_ordering_errors ... ok
test tests::verify_options_rejects_report_wide_without_report_mode ... ok
test tests::verify_options_requires_port_for_tcp_udp ... ok
test tests::validate_target_rejects_invalid_target ... ok
test tests::validate_target_accepts_ip_and_hostname ... ok

test result: ok. 16 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s

     Running tests\cli_tests.rs (target\debug\deps\cli_tests-0a025fc626855789.exe)

running 5 tests
test test_basic_execution ... ignored
test test_help_output ... ok
test test_version_output ... ok
test test_enhanced_ui_conflicts_with_report_mode ... ok
test test_enhanced_wrapper_conflicts_with_passthrough_override ... FAILED

failures:

---- test_enhanced_wrapper_conflicts_with_passthrough_override stdout ----

thread 'test_enhanced_wrapper_conflicts_with_passthrough_override' (5212) panicked at tests\cli_tests.rs:65:5:
assertion failed: stderr.contains("--trippy-flags cannot override windows-mtr enhanced UI wrapper settings")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_enhanced_wrapper_conflicts_with_passthrough_override

test result: FAILED. 3 passed; 1 failed; 1 ignored; 0 measured; 0 filtered out; finished in 21.25s

error: test failed, to rerun pass `-p windows-mtr --test cli_tests`
Error: Process completed with exit code 1.
```

--- a/src/main.rs
+++ b/src/main.rs
@@ -105,7 +105,11 @@ struct Cli {
     dns_cache_ttl: Option<u64>,
 
     /// Forward additional native trippy options verbatim
-    #[arg(long = "trippy-flags", value_name = "FLAGS")]
+    #[arg(
+        long = "trippy-flags",
+        value_name = "FLAGS",
+        allow_hyphen_values = true
+    )]
     trippy_flags: Option<String>,
 
     /// UI preset for interactive mode

